### PR TITLE
Laying foundation for YAML import tool

### PR DESCRIPTION
**Pull request by tomikazi**

### CODE REVIEW

File: pkg/loader/loader.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+// LoadTopology loads the specified YAML file and creates the prescribed simulated topology entities
+// using the fabric simulator API client.
+func LoadTopology(conn *grpc.ClientConn, topologyPath string) error {
+	topology := &Topology{}
+	if err := LoadTopologyFile(topologyPath, topology); err != nil {
+		return err
+	}
+
+	log.Infof("%+v", topology)
```
Comment (adibrastegarnia) [inline pkg/loader/loader.go:25]: Perhaps make this debug. I assume it is gonna be a big log

File: pkg/simulator/core.go [hunk 2]
```diff
@@ -15,26 +17,29 @@ import (
...
+	deviceSimulators map[simapi.DeviceID]*devsim.Simulator
+	linkSimulators   map[simapi.LinkID]*linksim.Simulator
+	hostSimulators   map[simapi.HostID]*hostsim.Simulator
 }
 
 // NewSimulation creates a new core simulation entity
 func NewSimulation() *Simulation {
 	return &Simulation{
-		deviceSimulators: make(map[simapi.DeviceID]*device.Simulator),
+		deviceSimulators: make(map[simapi.DeviceID]*devsim.Simulator),
```
Comment (adibrastegarnia) [inline pkg/simulator/core.go:28]: In ransim, we created simple data stores and an interface for each to make it easier to manage and interact with objects we want to create/get/update/delete. My suggestion is to separate stores from the actual logic but it is up to you.

File: pkg/loader/loader.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+	if err := createHosts(conn, topology.Hosts); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Create all simulated devices
+func createDevices(conn *grpc.ClientConn, devices []Device) error {
+	deviceClient := simapi.NewDeviceServiceClient(conn)
```
Comment (adibrastegarnia) [inline pkg/loader/loader.go:43]: Just for my information purposes: why internally we use gRPC to create devices?

**Review by adibrastegarnia — APPROVED** [review]

File: pkg/loader/loader.go (hunk 1 above)
Comment (tomikazi) [reply pkg/loader/loader.go:25]: Yes, I'll remove this... it's here purely for initial signal of activity. :)

File: pkg/simulator/core.go (hunk 2 above)
Comment (tomikazi) [reply pkg/simulator/core.go:28]: Possibly. This is equally to track the simulated entities and any simulation related processes/state. I'll leave it for now and may refactor later.

**Pull request merged**